org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/messages/progress/WorkDoneProgressNotification.java

```diff
@@ -0,0 +1,6 @@
+package org.eclipse.lsp4j.jsonrpc.messages.progress;
+
+public interface WorkDoneProgressNotification {
+
+	WorkDoneProgressKind getKind();
```

Comment:
We need a special Gson type adapter for this interface in order to parse incoming messages depending on the `kind` property. Maybe we can reuse this abstract class (or a variation of it): https://github.com/eclipse/sprotty-server/blob/master/org.eclipse.sprotty.server/src/main/java/org/eclipse/sprotty/server/json/PropertyBasedTypeAdapter.java